common/src/main/java/com/wynntils/screens/itemsharing/SavedItemsScreen.java

```diff
@@ -0,0 +1,528 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.itemsharing;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.items.FakeItemStack;
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.models.items.type.SavedItemStack;
+import com.wynntils.screens.base.TextboxScreen;
+import com.wynntils.screens.base.WynntilsContainerScreen;
+import com.wynntils.screens.base.widgets.InfoButton;
+import com.wynntils.screens.base.widgets.TextInputBoxWidget;
+import com.wynntils.screens.itemsharing.widgets.SavedCategoryButton;
+import com.wynntils.screens.itemsharing.widgets.SavedItemsButton;
+import com.wynntils.utils.EncodedByteBuffer;
+import com.wynntils.utils.MathUtils;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.mc.KeyboardUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import com.wynntils.utils.type.ErrorOr;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.inventory.ClickType;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import org.lwjgl.glfw.GLFW;
+
+public final class SavedItemsScreen extends WynntilsContainerScreen<SavedItemsMenu> {
+    private static final int ITEMS_PER_ROW = 7;
+    private static final int MAX_ITEMS = 49;
+    private static final int SCROLL_AREA_HEIGHT = 128;
+    private static final String DEFAULT_CATEGORY = "Uncategorized";
```

Comment:
Probably better referenced from the feature. Or even better. Move the whole storage of items to `ItemVaultService`, and store that constant there, along with the items. You could also make convenience methods in the model, without accessing the feature.

Reply:
Hm. I see now that it is stored in the encoding model, not the feature. Anyways, let's move away from that, and move to the service.